lib/src/presentation/view/card_list.dart

```diff
@@ -0,0 +1,341 @@
+import 'package:flutter/material.dart';
+import '../../core/util/card_constants.dart';
+import '../../data/model/card.dart';
+import '../bloc/card_bloc.dart';
+import 'card_page.dart';
+
+class CardList extends StatefulWidget {
+  final CardBloc cardBloc;
+
+  const CardList({
+    required this.cardBloc,
+    super.key,
+  });
+
+  @override
+  State<CardList> createState() => _CardListState();
+}
+
+class _CardListState extends State<CardList> {
+  final _totemUrl = kRacesTotemEndpointApi;
+  final _demonUrl = kRacesDemonEndPointApi;
+  final _murlocUrl = kRacesMurlocEndPointApi;
+  late String _currentUrl = _demonUrl;
+  late String actualPage = kCardListActualPage;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.cardBloc.initialize();
+  }
+
+  @override
+  void dispose() {
+    widget.cardBloc.dispose();
+    super.dispose();
+  }
+
+  Future<void> updateCurrentPage() async {
+    setState(() {
+      actualPage = widget.cardBloc.currentPage.toString();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Scaffold(
+        backgroundColor: Colors.black,
+        appBar: AppBar(
+          title: const Center(
+            child: Text(
+              appBarCardListTitle,
+              style: TextStyle(
+                fontSize: appBarCardListTitleFontSize,
+              ),
+            ),
+          ),
+          backgroundColor: Colors.black,
+        ),
+        endDrawer: Drawer(
+          backgroundColor: Colors.cyanAccent,
+          width: kCardListDrawerWidth,
+          child: ListView(
+            children: [
+              DrawerHeader(
+                decoration: BoxDecoration(
+                  color: Colors.purple,
+                  border: Border.all(
+                    width: kCardListDrawerHeaderWidth,
+                  ),
+                  borderRadius:
+                      BorderRadius.circular(kCardListDrawerHeaderWidth),
+                ),
+                curve: Curves.fastLinearToSlowEaseIn,
+                child: const Text(
+                  kCardListDrawerTitle,
+                  textAlign: TextAlign.center,
+                  style: TextStyle(
+                    fontSize: cardTextInfoFontSize,
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+              ),
+              ListTile(
+                title: Title(
+                  color: Colors.black,
+                  child: const Text(
+                    kCardListTotems,
+                    textAlign: TextAlign.center,
+                    style: TextStyle(
+                      fontSize: cardTextInfoFontSize,
+                      fontWeight: FontWeight.bold,
+                    ),
+                  ),
+                ),
+                onTap: () {
+                  _currentUrl = _totemUrl;
+                  widget.cardBloc.sinkCardList(
```

Comment:
Rename the method, i think is more appropriate to call it .callEndpoint(...) or getCardFromEndpoint(...)

Reply:
Done.